src/index.ts

```diff
@@ -58,11 +61,20 @@ const plugin: JupyterFrontEndPlugin<void> = {
         }
 
         const links = results.map(server => {
-          return URLExt.normalize(
-            `${PageConfig.getUrl({
-              workspace: PageConfig.defaultWorkspace
-            })}?token=${server.token}`
-          );
+          if (retroShell !== null) {
```

Comment:
Often we use the more canonical form (mostly cosmetics / readability):

```suggestion
          if (retroShell) {
```

Reply:
aaah, makes sense given this is typescript so it can't be something like `false` or `0`

Reply:
Fixed